vertx-mongo-client/src/test/java/io/vertx/ext/mongo/MongoClientTestBase.java

```diff
@@ -691,7 +696,12 @@ public void testReplaceUpsert2() {
         mongoClient.find(collection, new JsonObject(), onSuccess(list -> {
           assertNotNull(list);
           assertEquals(1, list.size());
-          assertEquals(id, list.get(0).getString("_id"));
+          Object id_value = list.get(0).getValue("_id");
+          if (id_value instanceof JsonObject) {
+            assertEquals(id, ((JsonObject) id_value).getString("$oid"));
+          } else {
+            assertEquals(id, (String) id_value);
+          }
```

Comment:
I've got no further code comments, good quality patch :-) - but @johnoliver
will need to check to make sure this isn't an alternative implementation of
a previously accepted patch.

On Monday, 14 September 2015, meggarr <notifications@github.com> wrote:

> In
> vertx-mongo-client/src/test/java/io/vertx/ext/mongo/MongoClientTestBase.java
> <https://github.com/vert-x3/vertx-mongo-client/pull/41#discussion_r39390711>
> :
>
> > @@ -691,7 +696,12 @@ public void testReplaceUpsert2() {
> >          mongoClient.find(collection, new JsonObject(), onSuccess(list -> {
> >            assertNotNull(list);
> >            assertEquals(1, list.size());
> > -          assertEquals(id, list.get(0).getString("_id"));
> > +          Object id_value = list.get(0).getValue("_id");
> > +          if (id_value instanceof JsonObject) {
> > +            assertEquals(id, ((JsonObject) id_value).getString("$oid"));
> > +          } else {
> > +            assertEquals(id, (String) id_value);
> > +          }
>
> Yes, I can wrap it to a method together with other comments if any.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vert-x3/vertx-mongo-client/pull/41/files#r39390711>.
>


-- 
Cheers, Martijn (Sent from Gmail Mobile)